servers/audio/effects/audio_effect_record.h

```diff
@@ -44,7 +44,7 @@ class AudioEffectRecord;
 class AudioEffectRecordInstance : public AudioEffectInstance {
 	GDCLASS(AudioEffectRecordInstance, AudioEffectInstance);
 	friend class AudioEffectRecord;
-	Ref<AudioEffectRecord> base;
+	Ref<WeakRef> base;
```

Comment:
Would it be possible to just use a WeakRef here? Pointing directly to the effect. It would also simplify the process logic.

Reply:
Upon a second look, it made sense to remove it altogether. Keeping a reference to the `AudioEffectRecord` even as a weak reference introduces a race condition that's hard to avoid -- that thread might attempt to destroy itself and call `Thread.wait_to_finish()` which it shouldn't. We need an different thread to make that call.

The main thread, which is the one `AudioServer` runs on, suffices. When the effect is removed from the bus, and if there's no other thread holding a reference to it, the `AudioEffecRecord`'s destructor will signal the thread that `AudioEffectRecordInstance` has running to finish.